### ISSUE TITLE
azurerm_log_analytics_solution unexpected status 200 with response: {"plan":{"name": . . .

### ISSUE THREAD

**Issue by mailbox171** [hashicorp/terraform-provider-azurerm#23312]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.1

### AzureRM Provider Version

3.73.0.

### Affected Resource(s)/Data Source(s)

azurerm_log_analytics_solution

### Terraform Configuration Files

```hcl
# Not possible to reproduce with minimal config


resource "azurerm_log_analytics_solution" "solution" {
  for_each = var.solution_names

  solution_name         = each.key
  location              = var.location
  resource_group_name   = var.resource_group_name
  workspace_resource_id = var.workspace_resource_id
  workspace_name        = var.workspace_name

  plan {
    publisher = var.publisher
    product   = "OMSGallery/${each.key}"
  }

  tags = var.tags
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: creating/updating Solution (Subscription: "***"
│ Resource Group Name: "rg-sdp-prod"
│ Solution Name: "ContainerInsights(log-prod)"): performing CreateOrUpdate: unexpected status 200 with response: {"plan":{"name":"ContainerInsights(log-prod)","publisher":"Microsoft","promotionCode":"","product":"OMSGallery/ContainerInsights"},"properties":{"workspaceResourceId":"/subscriptions/***/resourceGroups/rg-sdp-prod/providers/Microsoft.OperationalInsights/workspaces/log-prod","provisioningState":"Succeeded","creationTime":"Thu, 11 May 2023 10:11:26 GMT","lastModifiedTime":"Tue, 19 Sep 2023 07:50:54 GMT","containedResources":[],"workbookTemplates":[]},"location":"northeurope","tags":{"AppName":"ADAS - Advanced Driver Assistance System","Environment":"Production","StableforReservation":"No","WBS Code":"P19111"},"id":"/subscriptions/***/resourceGroups/rg-sdp-prod/providers/Microsoft.OperationsManagement/solutions/ContainerInsights(log-prod)","name":"ContainerInsights(log-prod)","type":"Microsoft.OperationsManagement/solutions"}
│ 
│   with module.loganalytics_solutions.azurerm_log_analytics_solution.solution["ContainerInsights"],
│   on .terraform/modules/loganalytics_solutions/loganalytics_solution/main.tf line 1, in resource "azurerm_log_analytics_solution" "solution":
│    1: resource "azurerm_log_analytics_solution" "solution" {
│ 
╵
Error: Process completed with exit code 1.
```

### Expected Behaviour

Just tags updated, as for PLAN below

```
# module.loganalytics_solutions.azurerm_log_analytics_solution.solution["ContainerInsights"] will be updated in-place
  ~ resource "azurerm_log_analytics_solution" "solution" {
        id                    = "/subscriptions/***/resourceGroups/rg-sdp-prod/providers/Microsoft.OperationsManagement/solutions/ContainerInsights(log-prod)"
      ~ tags                  = {
          - "AppName"              = "ADAS - Advanced Driver Assistance System" -> null
          - "Environment"          = "Production" -> null
          - "StableforReservation" = "No" -> null
          - "WBS Code"             = "P19111" -> null
        }
        # (5 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }
```

### Actual Behaviour

APPLY fails with error

```
╷
│ Error: creating/updating Solution (Subscription: "***"
│ Resource Group Name: "rg-sdp-prod"
│ Solution Name: "ContainerInsights(log-prod)"): performing CreateOrUpdate: unexpected status 200 with response: {"plan":{"name":"ContainerInsights(log-prod)","publisher":"Microsoft","promotionCode":"","product":"OMSGallery/ContainerInsights"},"properties":{"workspaceResourceId":"/subscriptions/***/resourceGroups/rg-sdp-prod/providers/Microsoft.OperationalInsights/workspaces/log-prod","provisioningState":"Succeeded","creationTime":"Thu, 11 May 2023 10:11:26 GMT","lastModifiedTime":"Tue, 19 Sep 2023 07:50:54 GMT","containedResources":[],"workbookTemplates":[]},"location":"northeurope","tags":{"AppName":"ADAS - Advanced Driver Assistance System","Environment":"Production","StableforReservation":"No","WBS Code":"P19111"},"id":"/subscriptions/***/resourceGroups/rg-sdp-prod/providers/Microsoft.OperationsManagement/solutions/ContainerInsights(log-prod)","name":"ContainerInsights(log-prod)","type":"Microsoft.OperationsManagement/solutions"}
│ 
│   with module.loganalytics_solutions.azurerm_log_analytics_solution.solution["ContainerInsights"],
│   on .terraform/modules/loganalytics_solutions/loganalytics_solution/main.tf line 1, in resource "azurerm_log_analytics_solution" "solution":
│    1: resource "azurerm_log_analytics_solution" "solution" {
│ 
╵
Error: Process completed with exit code 1.
```

### Steps to Reproduce

terraform plan
terraform apply

### Important Factoids

We provisioned in may with latest 3.x version of the provider available at that time. 
We had recently manually created a diagnostic setting in the AKS cluster using the workspace, but we have removed everything and restored the previous situation (all TF provisioned).

### References

No

**Comment by mailbox171** [comment]
Lookinh in Azure Portal at the  Activity log of ContainerInsights(log-prod) Solution, It seems that even if the TF PLAN shows only tags to be updated (which is CORRECT), a new OMS Solution is created everytime we TF APPLY.
I attach the JSON description.

[Create-new-OMS-solution.json.txt](https://github.com/hashicorp/terraform-provider-azurerm/files/12663996/Create-new-OMS-solution.json.txt)

**Comment by ziyeqf** [comment]
Hi @mailbox171, thanks for reporting.

It was found the API definition for the `update` operation has some issue, An issue on `azure-rest-api-specs` has been raised(https://github.com/Azure/azure-rest-api-specs/issues/25866), once it fixed we can fix it in our provider.

For any other questions please leave comments.

**Comment by mailbox171** [comment]
Thank you very much!

Looks like the issue is there in provider 3.73.0, but was not there in previsous version.
Do you have an idea about when it was introduced?

**Comment by ziyeqf** [comment]
This should be introduced on this PR (https://github.com/hashicorp/terraform-provider-azurerm/pull/23088) which affected 3.72.0 and 3.73.0.

And just found it could be fixed in our provider, Have submitted a PR (#23333 ) for it, once it merged & released, it should be fixed. You can subscribe that PR to keep track on it.

**Comment by mailbox171** [comment]
Thanks a lot!

How can I use the fix? I must wait 3.75 is released?

**Comment by ziyeqf** [comment]
> Thanks a lot!
> 
> How can I use the fix? I must wait 3.75 is released?

Per the tag of the PR, it will be release on v3.74.0 which is expected to be released this week.
